src/qs_energy.F

```diff
@@ -95,7 +95,11 @@ SUBROUTINE qs_energies(qs_env, consistent_energies, calc_forces)
 
          IF (PRESENT(consistent_energies)) THEN
             IF (consistent_energies) THEN
-               CALL qs_ks_update_qs_env(qs_env, calculate_forces=.FALSE., just_energy=.TRUE.)
+               IF (dft_control%do_rtp_periodic) THEN
```

Comment:
`dft_control%do_rtp_periodic` is just a synonym of the variable `dft_control%rtp_control%periodic`. It may be a better option to use a local logical variable within the subroutine `qs_energies()` instead of introducing yet another component of the `dft_control` derived type :

```
LOGICAL :: do_rtp_periodic
...
do_rtp_periodic = .FALSE.
IF (ASSOCIATED(dft_control%rtp_control)) &
   do_rtp_periodic = dft_control%rtp_control%periodic

CALL qs_ks_update_qs_env(qs_env, calculate_forces=.FALSE., just_energy=.NOT. do_rtp_periodic)
```

Reply:
I faintly remember trying that first, but the rtp_control was not initiated for every run type, I think

Reply:
In case there is no &REAL_TIME_PROPAGATION section in the input file, `rtp_control` is just a non-associated (NULL) pointer.